spring-integration-core/src/main/java/org/springframework/integration/gateway/GatewayProxyFactoryBean.java

```diff
@@ -968,31 +956,27 @@ private void channels(@Nullable String requestChannelName, @Nullable String repl
 
 	private void timeouts(@Nullable Expression requestTimeout, @Nullable Expression replyTimeout,
 			GatewayMethodInboundMessageMapper messageMapper, MethodInvocationGateway gateway) {
-		if (requestTimeout == null) {
-			gateway.setRequestTimeout(-1);
-		}
-		else if (requestTimeout instanceof ValueExpression) {
-			Long timeout = requestTimeout.getValue(Long.class);
-			if (timeout != null) {
-				gateway.setRequestTimeout(timeout);
+		if (requestTimeout != null) {
```

Comment:
We still need a null check here and use the default; `GenericMessagingTemplate` has `-1` as its default.

Reply:
Sure! See `MessagingGatewaySupport`:
```
private long replyTimeout = IntegrationContextUtils.DEFAULT_TIMEOUT;
...
template.setReceiveTimeout(this.replyTimeout);
```
So, if we don't call `setReplyTimeout()`, then that default value is used in the `GenericMessagingTemplate`.

Reply:
Ah - ok.